### DISCUSSION TITLE
How to specify port in environment variables?

### DISCUSSION THREAD

**Discussion by anubi** [requarks/wiki#5385] [Help / Questions] [answered]
I am deploying the wiki to Azure via an "azure container instance" (ACI), it works great, but the problem is I want the port to be 80, not 3000. I can /bin/bash into the running image and change config.yml's port and restart the container, but that doesn't actually change the port. I am not using docker-compose, so I can't forward the port when creating an ACI. I have to specify an environment variable, but "Port" doesn't seem to work? I can obviously change the database port with "DB_PORT", but the site itself is stubbornly fixed on 3000? Am I doing something obviously wrong? Any ideas? I would rather not provision a whole VM just to run this thing.

**Comment by NGPixel** [comment]
Port 3000 is the internal docker container port. You should be able to map 3000 to the port of your choice.

**Comment by anubi** [comment]
Thanks, but specifically as I said, ACIs do not let you map 80:3000 or 443:3443. Ultimately, I had to create a VM in azure to be able to host it. Which is really too bad because the way I would like to do it would be more elegant.

If you're curious how you can recreate the problem, play around with the following

(https://docs.docker.com/cloud/aci-integration/)
1. use standard docker-compose for the wiki
2. in docker desktop, do: 
3. docker login azure
4. docker context create aci myacicontext
5. docker context use myacicontext
6. docker compose up (notice the lack of -)

This will run all the necessary azure commands behind the scenes to create your infrastructure on the fly.

It will fail to deploy, saying mapping 80:3000 is not supported. You have to supply 80:80 or 443:443, which makes you unable to contact the wiki from outside your container. And there seems to be no straight forward way for me to change the port from 3000 for a compose.

**Reply by NGPixel** [reply]
Seems like a major omission from Azure to not support such a basic feature like port mapping...

A workaround would be to extend the regular wiki.js docker image with your own configuration, e.g.:

```dockerfile
FROM ghcr.io/requarks/wiki:2

COPY --chown=node:node YOUR-CUSTOM-CONFIG.yml ./config.yml

EXPOSE 80
EXPOSE 443
```

Replace `YOUR-CUSTOM-CONFIG.yml` with your own [config.yml](https://github.com/requarks/wiki/blob/main/dev/build/config.yml) with the ports set to 80 / 443.

**Reply by NGPixel** [reply]
or another "hacky" workaround is to specify both these environment variables:

```toml
HEROKU=1
PORT=80
```

This will bypass the value from the config.yml and use the port defined in variable `PORT`

**Reply by moedeveloper** [reply]
is thats what you mean? where the 9910 is the internal container port.

![image](https://user-images.githubusercontent.com/16238332/208940936-de038eb3-1205-4b83-a4a7-d67cdc2d1818.png)

**Comment by anubi** [comment, marked as answer]
Oh, thank you for the second suggestion! I will try it. I did want to avoid creating a dockerfile if necessary, but you're right maybe that's the more sensible option...
And yeah, I don't know why azure only supports 80/443 in ACI contexts, it seems like a random thing they chose for "security" reasons.

**Comment by anubi** [comment]
It ended up working out that way as you suggested
